npc/re/merchants/enchan_mora.txt

```diff
@@ -580,16 +580,16 @@ mora,152,97,5	script	Guardian of Power#pa082	4_F_DOGTRAVELER,{
 	next;
 	switch(.@job) {
 	case 0: // Rune Knight
-		setarray .@items[0],2475,2476,2574,2575;
-		.@i = select("Cancel", "Ur's Greaves (Shoes)", "Peuz's Greaves (Shoes)", "Ur's Manteau (Garment)", "Peuz's Manteau (Garment)")-2;
+		setarray .@items[0],2475,2476,2574,2575,15036,15037,2883,2884;
+		.@i = select("Cancel", "Ur's Greaves (Shoes)", "Peuz's Greaves (Shoes)", "Ur's Manteau (Garment)", "Peuz's Manteau (Garment)", "Ur's Plate (Armor", "Peuz's Plate (Armor)", "Ur's Seal (Accessory)", "Peuz's Seal (Accessory)")-2;
```

Comment:
You're missing a `)` after `Ur's Plate (Armor`

also please add these in the correct order `Ur's` set then `Peuz's`, this is the list from kRO for reference

```
"우르의그리브", "우르의망토", "우르의인장", "우르의플레이트", "페오즈의그리브", "페오즈의망토", "페오즈의인장", "페오즈의플레이트"
```

Reply:
Will do the changes. Thank you for reviewing.

Do you mean that I should enumerate Ur's set first, then Peuz's set next? And not to use the korean translation? Clarifying to make sure.

Reply:
What I meant is: make it in the same order as in the list I gave :)

Reply:
Okay. Will do. :D

Reply:
instead of hardcode the menu, maybe just let it generate based on the name in item_db?

replace
```
	switch(.@job) {
	case 0: // Rune Knight
		setarray .@items[0],2475,2574,2883,15036,2575,2476,2884,15037;
		.@i = select("Cancel", "Ur's Greaves (Shoes)",  "Ur's Manteau (Garment)", "Ur's Seal (Accessory)",  "Ur's Plate (Armor)", "Peuz's Greaves (Shoes)", "Peuz's Manteau (Garment)",  "Peuz's Seal (Accessory)", "Peuz's Plate (Armor)" )-2;
		break;
	case 1: // Guillotine Cross
		setarray .@items[0],2477,2577,2886,15038,2478,2578,2887,15039;
		.@i = select("Cancel", "Sapha Shoes (Shoes)",  "Sapha Hood (Garment)",  "Sapha Ring (Accessory)", "Sapha's Cloth (Armor)", "Nab Shoes (Shoes)", "Nab Hood (Garment)", "Nab Ring (Accessory)", "Nab's Cloth (Armor)" )-2;
		break;
	case 2: // Ranger
		setarray .@items[0],2479,2580,2890,15042,2480,2581,2891,15043;
        .@i = select("Cancel", "White Wing Boots (Shoes)",  "White Wing Manteau (Garment)", "White Wing Brooch (Accessory)", "White Wing Suit (Armor)", "Black Wing Boots (Shoes)","Black Wing Manteau (Garment)",  "Black Wing Brooch (Accessory)", "Black Wing Suit (Armor)" )-2;
		break;
	}
```
with
```
	switch(.@job) {
	case 0: // Rune Knight
		setarray .@items[0],2475,2574,2883,15036,2575,2476,2884,15037;
		break;
	case 1: // Guillotine Cross
		setarray .@items[0],2477,2577,2886,15038,2478,2578,2887,15039;
		break;
	case 2: // Ranger
		setarray .@items[0],2479,2580,2890,15042,2480,2581,2891,15043;
		break;
	}
	.@menu$ = "Cancel";
	for (.@items_size = getarraysize(.@items); .@i < .@items_size; .@i++)
		.@menu$ = .@menu$ + ":" + getitemname(.@items[.@i]) + "("+F_GetArmorType(.@items[.@i])+")";
	.@i = select(.@menu$) - 2;
```

Reply:
@darrensapalo it's existing function, you can find it here https://github.com/HerculesWS/Hercules/blob/stable/npc/other/Global_Functions.txt#L293-L369